lib/vrl/stdlib/src/parse_json.rs

```diff
@@ -162,5 +263,41 @@ mod tests {
                 .add_array(Collection::from_unknown(inner_kind()))
                 .add_object(Collection::from_unknown(inner_kind())),
         }
+
+        max_depth {
+            args: func_args![ value: r#"{"top_layer": {"layer_one": "finish", "layer_two": 2}}"#, max_depth: 1],
+            want: Ok(value!({ top_layer: r#"{"layer_one": "finish", "layer_two": 2}"# })),
+            tdef: type_def(),
+        }
+
+        max_depth_array {
+            args: func_args![ value: r#"[{"top_layer": {"next_layer": ["finish"]}}]"#, max_depth: 2],
+            want: Ok(value!([{ top_layer: r#"{"next_layer": ["finish"]}"# }])),
+            tdef: type_def(),
+        }
+
+        max_depth_exhausted {
+            args: func_args![ value: r#"{"top_layer": {"layer_one": "finish", "layer_two": 2}}"#, max_depth: 10],
+            want: Ok(value!({ top_layer: {layer_one: "finish", layer_two: 2} })),
+            tdef: type_def(),
+        }
```

Comment:
None of these tests appear to actually test what happens when the `max_depth` is exhausted.

Reply:
Yeah, I meant that the max_depth value exceeds the number of the JSON document. I will fix the naming.